Update exception-filters.md (#2548)
nestjs/docs.nestjs.com: content/exception-filters.md

> info **Hint** All exception filters should implement the generic `ExceptionFilter<T>` interface. This requires you to provide the `catch(exception: T, host: ArgumentsHost)` method with its indicated signature. `T` indicates the type of the exception.

> If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.

Reviewer: I believe it should be "you should use" instead of "you can use"

> If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.

Reviewer 2: ```suggestion
> warning **Warning** If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.
```

> you can use `response.send()` instead of `response.json()` when using `platform-fastify`, and use `import { FastifyReply as Response, FastifyRequest as Request } from 'fastify';` instead of `import { Request, Response } from 'express';`

Reviewer 3: With the way Dgeni (our markdown to angular component generator) works, you can't have `{` or `}` in the direct text (i.e. not a codeblock) without properly escaping it. If  recall, you have to do `{{ '{' }}` and `{{ '}' }}` to add them.

I think rather  we can re-write this sentence to be something like

```suggestion
If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.
```

and then have the example below to show the import of the types and .usage

like this:

```typescript
@@filename(http-exception.filter)
import { ExceptionFilter, Catch, ArgumentsHost, HttpException } from '@nestjs/common';
import { FastifyReply as Response, FastifyRequest as Request } from 'fastify';

@Catch(HttpException)
export class HttpExceptionFilter implements ExceptionFilter {
  catch(exception: HttpException, host: ArgumentsHost) {
    const ctx = host.switchToHttp();
    const response = ctx.getResponse<Response>();
    const request = ctx.getRequest<Request>();
    const status = exception.getStatus();

    response
      .status(status)
      .send({
        statusCode: status,
        timestamp: new Date().toISOString(),
        path: request.url,
      });
  }
}
@@switch
import { Catch, HttpException } from '@nestjs/common';

@Catch(HttpException)
export class HttpExceptionFilter {
  catch(exception, host) {
    const ctx = host.switchToHttp();
    const response = ctx.getResponse();
    const request = ctx.getRequest();
    const status = exception.getStatus();

    response
      .status(status)
      .send({
        statusCode: status,
        timestamp: new Date().toISOString(),
        path: request.url,
      });
  }
}
```

> > info **Hint** you must install `fastify`

Reviewer: We can drop this info as in this page we do not pretend to teach on how to use the fastify adapter

Author: This infor is only used to import Request and Response from the factory
It has better type hints, which are optional

> 
> like this:
> 
> ```typescript
> @@filename(http-exception.filter)
> import { ExceptionFilter, Catch, ArgumentsHost, HttpException } from '@nestjs/common';
> import { FastifyReply as Response, FastifyRequest as Request } from 'fastify';
> 
> @Catch(HttpException)
> export class HttpExceptionFilter implements ExceptionFilter {
>   catch(exception: HttpException, host: ArgumentsHost) {
>     const ctx = host.switchToHttp();
>     const response = ctx.getResponse<Response>();
>     const request = ctx.getRequest<Request>();
>     const status = exception.getStatus();
> 
>     response
>       .status(status)
>       .send({
>         statusCode: status,
>         timestamp: new Date().toISOString(),
>         path: request.url,
>       });
>   }
> }
> @@switch
> import { Catch, HttpException } from '@nestjs/common';
> 
> @Catch(HttpException)
> export class HttpExceptionFilter {
>   catch(exception, host) {
>     const ctx = host.switchToHttp();
>     const response = ctx.getResponse();
>     const request = ctx.getRequest();
>     const status = exception.getStatus();
> 
>     response
>       .status(status)
>       .send({
>         statusCode: status,
>         timestamp: new Date().toISOString(),
>         path: request.url,
>       });
>   }
> }
> ```
> 
> > info **Hint** you must install `fastify`
> 

Reviewer 2: ```suggestion
```

> info **Hint** All exception filters should implement the generic `ExceptionFilter<T>` interface. This requires you to provide the `catch(exception: T, host: ArgumentsHost)` method with its indicated signature. `T` indicates the type of the exception.

> > warning **Warning** If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.

Reviewer 2: ```suggestion
> warning **Warning** If you are using `@nestjs/platform-fastify` you can use `response.send()` instead of `response.json()`. Don't forget to import the correct types from `fastify`.

```
